commit 2d142c09022d1b03479b7e22dd5d49a5573484ea (refs/changes/86/1686/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-17T21:00:18+00:00 (2 years, 7 months ago)
    
    CNS-198 Peer version failure neglects to call callback; leaks redis handle
    Reviewed by: Richard Kiene <richard.kiene@joyent.com>

diff --git a/lib/dns-server.js b/lib/dns-server.js
index 72e790d..fca411d 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -1083,6 +1083,7 @@ DNSServer.prototype.handleTransfer = function (q, z, cb) {
 						log.debug(perr,
 						    'unable to detect peer ' +
 						    'version');
+						ccb(perr);
 						return;
 					}
 					_.redis.set('peer:' +
